internal/runtime/client/client.go

```diff
@@ -488,8 +488,13 @@ func httpCall(ctx context.Context, request, response runtime.Object, opts *httpC
 	})
 
 	resp, err := client.Do(httpRequest)
+	respStatus := "Unknown"
+
 	// Create http request metric.
-	runtimemetrics.RequestsTotal.Observe(httpRequest, resp, opts.hookGVH, err)
+	defer func() {
+		runtimemetrics.RequestsTotal.Observe(httpRequest, resp, opts.hookGVH, err, respStatus)
+	}()
```

Comment:
Does this work if the below cases create a new error (e.g. the `	if err := json.NewDecoder(resp.Body).Decode(responseLocal); err != nil {
`)?

Or do we have to define `var err error` first and always re-use that?

Reply:
Maybe it is possible to move the Observe method up to the function which calls `httpCall` so we don't have to use defer here?

Reply:
This error value is not used in labels in `Observe` method - just to check what to set in `code` label

https://github.com/MaxFedotov/cluster-api/blob/4b9dbdfec217947849522b2878c8d198355dc56d/internal/runtime/metrics/metrics.go#L69-L79

Reply:
> Maybe it is possible to move the Observe method up to the function which calls `httpCall` so we don't have to use defer here and only typecast once.

The problem is that in this case, all other people who will use `httpCall` method in the future must also not forget to add `Observe` method to their code, which can lead to missing metrics.

Reply:
> This error value is not used in labels in `Observe` method - just to check what to set in `code` label
> 
> https://github.com/MaxFedotov/cluster-api/blob/4b9dbdfec217947849522b2878c8d198355dc56d/internal/runtime/metrics/metrics.go#L69-L79

Ah right 👍  should we document that in the comment for the Observe function :-)